app/models/product.rb

```diff
@@ -1,12 +1,34 @@
+class UrlValidator < ActiveModel::EachValidator
```

Comment:
create a new folder `validators` and move it there.

Reply:
done